src/StreamJsonRpc/Reflection/MessageFormatterProgressTracker.cs

```diff
@@ -0,0 +1,168 @@
+        public readonly Dictionary<long, ProgressParamInformation> ProgressMap = new Dictionary<long, ProgressParamInformation>();
+
+        /// <summary>
+        /// Special method name for progress notification.
+        /// </summary>
+        internal const string ProgressRequestSpecialMethod = "$/progress";
+
+        /// <summary>
+        /// Object used to lock the access to <see cref="RequestProgressMap"/> and <see cref="ProgressMap"/>.
+        /// </summary>
+        internal readonly object ProgressLock = new object();
+
+        /// <summary>
+        /// Dictionary used to map the outbound request id to their progress id token so that the progress objects are cleaned after getting the final response.
+        /// </summary>
+        internal readonly Dictionary<long, long> RequestProgressMap = new Dictionary<long, long>();
+
+        /// <summary>
+        /// Gets or sets the the next id value to assign as token for the progress objects.
+        /// </summary>
+        internal long NextProgressId { get; set; }
+
+        /// <summary>
+        /// Gets or Sets the id of the request currently being serialized so the converter can use it to create the request-progress map.
+        /// </summary>
+        internal long? RequestIdBeingSerialized { get; set; }
+
+        /// <summary>
+        /// Converts given <see cref="Type"/> to its <see cref="IProgress{T}"/> type.
+        /// </summary>
+        /// <param name="objectType">The type which may implement <see cref="IProgress{T}"/>.</param>
+        /// <returns>The <see cref="IProgress{T}"/> from given <see cref="Type"/> object, or <c>null</c>  if no such interface was found in the given <paramref name="objectType" />.</returns>
+        public static Type FindIProgressOfT(Type objectType)
+        {
+            Requires.NotNull(objectType, nameof(objectType));
+
+            if (objectType.IsConstructedGenericType && objectType.GetGenericTypeDefinition().Equals(typeof(IProgress<>)))
+            {
+                return objectType;
+            }
+
+            return objectType.GetTypeInfo().GetInterfaces().FirstOrDefault(i => i.IsConstructedGenericType && i.GetGenericTypeDefinition() == typeof(IProgress<>));
+        }
+
+        /// <summary>
+        /// Saves a <see cref="ProgressParamInformation"/> instance obtained from the given <see cref="object"/> into <see cref="ProgressMap"/>.
+        /// </summary>
+        /// <param name="value">The object which should implement <see cref="IProgress{T}"/> to create the <see cref="ProgressParamInformation"/> instance.</param>
+        /// <returns>The assigned <see cref="long"/> progres ID.</returns>
+        public long AddProgressObjectToMap(object value)
+        {
+            Requires.NotNull(value, nameof(value));
+
+            if (this.RequestIdBeingSerialized == null)
+            {
+                throw new NotSupportedException("IProgress<T> objects should not be part of any response or notification.");
```

Comment:
>"IProgress<T> objects should not be part of any response or notification." [](start = 48, length = 74)

This should be moved to a localized string along with the others you've already done.

Reply:
Moved in latest update.